### PR TITLE
Balder/use searchable copies for coverage estimate

### PR DESCRIPTION
**Pull request by baldersheim**
@geirst  PR

### CODE REVIEW

**Comment by baldersheim** [comment]
Only config production and FNET_Search::CheckCoverage has semantic impact.
The rest is just code health.

File: searchcore/src/vespa/searchcore/fdispatch/search/fnet_search.cpp [hunk 1]
```diff
@@ -886,7 +886,7 @@ FastS_FNET_Search::CheckCoverage()
             cntNone++;
         }
     }
-    if ((cntNone > 0) && (cntNone != _nodes.size())) {
+    if ((cntNone >= _dataset->getSearchableCopies()) && (cntNone != _nodes.size())) {
         activeDocs += cntNone * activeDocs/(_nodes.size() - cntNone);
```
Comment (geirst) [inline searchcore/src/vespa/searchcore/fdispatch/search/fnet_search.cpp]: We must adjust cntNode with (searchableCopies - 1)

**Review by geirst — COMMENTED** [review]
Please look at my comments

File: config-model/src/main/java/com/yahoo/vespa/model/content/ContentSearchCluster.java [hunk 2]
```diff
@@ -263,8 +263,11 @@ public boolean usesHierarchicDistribution() {
     }
 
     public void handleRedundancy(Redundancy redundancy) {
-        if (usesHierarchicDistribution()) {
-            indexedCluster.setMaxNodesDownPerFixedRow((redundancy.effectiveFinalRedundancy() / groupToSpecMap.size()) - 1);
+        if (hasIndexedCluster()) {
+            if (usesHierarchicDistribution()) {
+                indexedCluster.setMaxNodesDownPerFixedRow((redundancy.effectiveFinalRedundancy() / groupToSpecMap.size()) - 1);
+            }
+            indexedCluster.setSearchableCopies(redundancy.readyCopies());
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply config-model/src/main/java/com/yahoo/vespa/model/content/ContentSearchCluster.java]: Correct, fixed

File: searchcore/src/vespa/searchcore/fdispatch/search/fnet_search.cpp (hunk 1 above)
Comment (baldersheim) [reply searchcore/src/vespa/searchcore/fdispatch/search/fnet_search.cpp]: Fixed

**Comment by baldersheim** [comment]
@geirst Now it should be completed.

**Review by geirst — APPROVED** [review]
👍

**Pull request merged**